shell/plugins/steve/worker/resourceCaches/resourceCache.js

```diff
@@ -0,0 +1,471 @@
+
+          return name;
+        }
+      },
+      {
+        namespacedNameSort: (resource) => {
+          const namespacedName = resource.namespacedName;
+
+          return sortableNumericSuffix(namespacedName).toLowerCase();
+        }
+      },
+      // groupByLabel: resource => '',
+      { stateObj: resource => resource.metadata?.state },
+      { state: resource => resource.metadata?.state?.name || 'unknown' },
+      { stateDisplay: resource => stateDisplay(resource.state) },
+      {
+        stateColor:   (resource) => {
+          return colorForState.call(
+            resource,
+            resource.state,
+            resource.stateObj?.error,
+            resource.stateObj?.transitioning
+          );
+        }
+      },
+      // stateBackground: resource => '',
+      // stateIcon: resource => '',
+      { stateSort: resource => stateSort(resource.stateColor, resource.stateDisplay) },
+      {
+        stateDescription: (resource) => {
+          const stateObj = resource.stateObj;
+
+          if (stateObj) {
+            const { transitioning: trans, error, message } = resource.stateObj;
+
+            return trans || error ? ucFirst(message) : '';
+          }
+        }
+      },
+      // details: resource => [];
+      // t: resource => {} // it'd be nice to get this one in but have to decouple from store first
+
+    ];
+
+    // ToDo: the constructor itself should kick off it's initial and potentially only query, the worker can observe when it's ready because it's subscription will hit "WATCHING" status
+  }
+
+  setStatus(status) {
+    this.status = status;
+
+    return this;
+  }
+
+  getStatus() {
+    return this.status;
+  }
+
+  _addExtraField(appendedResource, extraField, index) {
+    const [fieldName] = Object.keys(extraField);
+    const fieldFunction = extraField[fieldName];
+
+    return {
+      ...appendedResource,
+      extraFields: {
+        ...appendedResource.extraFields,
+        [fieldName]: fieldFunction(appendedResource)
+      },
+      [fieldName]: fieldFunction(appendedResource)
+    };
+  }
+
+  // otw: it turns the cache into an array for packaging it up in various ways
+  get asList() {
+    return Object.keys(this._resourceHashes).map(key => this._cache[key]);
+  }
+
+  get #totalResources() {
+    return Object.keys(this._resourceHashes).length;
+  }
+
+  // otw: gets the resource collection from the API
+  async makeRequest() {
+    const requestParams = {
+      url:     this.#schema?.links?.collection,
+      headers: { Accept: 'application/json' } // ToDo: should probably make this more complete but it works for now and seems faster
+    };
+
+    if (this.watchStatus() !== WATCHING) {
+      if (this._requesting) {
+        await waitFor(() => {
+          return !this._requesting;
+        });
+
+        return this;
+      } else {
+        this._requesting = true;
+        const { resourceResponse } = await allHash({
+          resourceResponse: axios(requestParams),
```

Comment:
Making the request outside of the store's `request` action will skip any special cases there.  At the moment, for just the steve like stores, and ignoring 401 handling, streaming and pagination, it works. 

However if we look to do other things, like support this for other types of store or which to filter by namespace it will require duplicated work